lib/TTT/minimax.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Minimax do
+  def minimax(state) do
+    with :game_over <- Status.over(State.get_board(state)) do
+      score_move(state, Status.winner(state.board))
+    else
+      :game_in_progress ->
+        available_moves = Map.keys(Player.available_moves(state))
+
+        scored_moves =
+          Enum.map(available_moves, fn position ->
+            new_state = Player.place_move(position, state)
+            %{player: state.current_player, position: position, score: minimax(new_state).score}
+          end)
+
+        optimal_move(scored_moves, state)
+    end
+  end
+
+  def score_move(state, winner) do
+    cond do
+      state.player == winner ->
```

Comment:
(Not sure if this would work, but could you use a case or check based on the value of winner?)

Reply:
you could but you'd have to create the variables first to match against winner because you can't invoke remote function in matches (like what I did in `optimal_move`, but I'm not sure if there's a preference or if one is more idiomatic)

Reply:
Could you deconstruct state and use variable pattern matching in the function?

I'm not 100% on what's better/easily possible, but this works